usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -283,34 +294,80 @@ def populate_response(self, results: list, has_next: bool) -> dict:
+            d = datetime.strptime(date, "%Y-%m-%d")
+            date = int(d.timestamp() * 1000)
+        return date
+
     def query_elasticsearch(self) -> list:
         filter_query = QueryWithFilters.generate_awards_elasticsearch_query(self.filters)
         sort_field = self.get_elastic_sort_by_fields()
         sorts = [{field: self.pagination["sort_order"]} for field in sort_field]
-        search = (
-            (
+        record_num = (self.pagination["page"] - 1) * self.pagination["limit"]
+
+        if (
+            self.last_record_unique_id is None and self.last_record_sort_value is None
+        ) or record_num >= settings.ES_AWARDS_MAX_RESULT_WINDOW:
+            sorts.append({"total_obligation": "desc"})
+            sorts.append({"award_id": self.pagination["sort_order"]})
+
+        if record_num >= settings.ES_AWARDS_MAX_RESULT_WINDOW and (
+            self.last_record_unique_id is None and self.last_record_sort_value is None
+        ):
+            logger.warning(
+                "WARNING: Jumping to page {page} with page size {limit}. First record number: {record}. Retrieving records past {es_limit} records will have a performance hit when using Elasticsearch".format(
+                    page=self.pagination["page"],
+                    limit=self.pagination["limit"],
+                    record=self.pagination["lower_bound"],
+                    es_limit=settings.ES_AWARDS_MAX_RESULT_WINDOW,
+                )
+            )
+            self.pagination["upper_bound"] = record_num
+            self.pagination["lower_bound"] = record_num - 1
+            queryset = self.construct_queryset()
+            if len(queryset) != 1:
+                return {}
+            results = []
+            results = [
+                self.date_to_epoch_millis(queryset[0].get(self.get_sort_by_fields()[0])),
+                float(queryset[0].get("total_obligation")),
+                queryset[0].get("award_id"),
+            ]
+            search = (
                 AwardSearch()
                 .filter(filter_query)
                 .sort(*sorts)
-                .extra(search_after=[self.last_value, self.last_id])[0 : self.pagination["limit"]]
+                .extra(search_after=[*results])[0 : self.pagination["limit"]]
             )
-            if self.last_value and self.last_id
-            else (
+            response = search.handle_execute()
+            return response
+        search = (
+            (
                 AwardSearch()
                 .filter(filter_query)
-                .sort(*sorts)[
-                    ((self.pagination["page"] - 1) * self.pagination["limit"]) : (
-                        ((self.pagination["page"] - 1) * self.pagination["limit"]) + self.pagination["limit"]
-                    )
-                ]
+                .sort(*sorts)
+                .extra(
+                    search_after=[
+                        self.date_to_epoch_millis(self.last_record_sort_value),
```

Comment:
So in this case, the api user has specified the `last_record_sort_value` directly -- effectively asking to jump to a _that_ page. If this is a new field in the API response (especially if we're going to just get rid of it soon), could you just enforce that they have to provide its value in epoch millis?

I'm also find leaving it as is. Seems to be a nice convenience to make the api call easier. But if the date conversion stuff gets messy, maybe just fallback to expecting it in millis.